### DISCUSSION TITLE
Need clarity on network interface configuration

### DISCUSSION THREAD

**Discussion by vbhurangi** [rtbrick/bngblaster#174] [Q&A] [answered]
Hi,
I am able to bring up subscribers with access interface.
Not able to bring up session or  generate traffic using network interface.

My virtual bng router is configured to accept pppoe clients with any vlan-id.

I have configure two interfaces (access and network) on ubuntu-vm( hosting bng-blaster) which are directly connected to virtual bng router. Also attached vlan interface to each interface for vlan traffic.

My issue is whenever I start the bng-blaster for session or network traffic then only ARP (network interface ip asks for Mac address of gateway) packets broadcasted  . 

  My configuration file looks like this:
  
  {
    "interfaces": {
        "rx-threads": 5,
        "tx-threads": 5,
        "io-slots": 16384,
        "network": {
            "interface": "vlan90-if",
            "address": "100.127.6.6/30",
            "gateway": "100.127.6.5",
            "vlan": 90
        },
        "access": [
            {
                "interface": "vlan100-if",
                "type": "pppoe",
                "inner-vlan": 100,
                "stream-group-id": 1,
                "authentication-protocol": "PAP"
            }
        ]
    },
    "session-traffic": {
            "ipv4-pps": 4500,
            "ipv6-pps": 0,
            "ipv6pd-pps": 0
    },
    "pppoe": {
         "reconnect": true,
         "discovery-timeout": 30,
         "discovery-retry": 10
    },
    "ppp": {
        "mru": 1492,
        "authentication": {
            "username":"vbhurangi",
            "password":"brasvmx",
            "timeout": 5,
            "retry": 30
     }
},
    "dhcpv6": {
        "enable": false
    },
    "session-traffic": {
        "ipv4-pps": 1
    }
}

Below is output from bngblaster:-

F1: Select View  F7/F8: Start/Stop Traffic  F9: Terminate Sessions                        Apr 27 11:47:37.079202 Warning: Interfaces must not have an IP address configured in the host OS!
F2: Network Interface  Left/Right: Access Interface                                       Apr 27 11:47:37.079462 Warning: IP address fe80::5403:3ff:fe01:42bc on interface vlan100-if is conflicting!
                                                                                                              Apr 27 11:47:37.214284 Warning: IP address fe80::5403:3ff:fe01:42b7 on interface vlan90-if is conflicting!
      ____   __   ____         _        __                                  ,/                    Apr 27 11:47:37.376813 pcap file test.pcap opened
     / __ \ / /_ / __ ) _____ (_)_____ / /__                              ,'/               Apr 27 11:47:37.377467 Opened control socket run.sock
    / /_/ // __// __  |/ ___// // ___// //_/                            ,' /                      Apr 27 11:47:38.377032 Resolve network interfaces
   / _, _// /_ / /_/ // /   / // /__ / ,<                             ,'  /_____,                 Apr 27 11:47:39.377619 Resolve network interfaces
  /_/ |_| \__//_____//_/   /_/ \___//_/|_|                          .'____    ,'           Apr 27 11:47:40.380749 Resolve network interfaces
      ____   _   _  ______   ____   _               _                    /  ,'                Apr 27 11:47:41.381690 Resolve network interfaces
     / __ ) / | / // ____/  / __ ) / /____ _ _____ / /_ ___   ____      / ,'         Apr 27 11:47:42.382492 Resolve network interfaces
    / __  |/  |/ // / __   / __  |/ // __ `// ___// __// _ \ / ___/    /,'                   Apr 27 11:47:43.383206 Resolve network interfaces
   / /_/ // /|  // /_/ /  / /_/ // // /_/ /(__  )/ /_ /  __// /       /                            Apr 27 11:47:44.387127 Resolve network interfaces
  /_____//_/ |_/ \____/  /_____//_/ \__,_//____/ \__/ \___//_/                   Apr 27 11:47:45.388005 Resolve network interfaces
                                                                                                                 Apr 27 11:47:46.388834 Resolve network interfaces
Test Duration: 10s                                                                                  Apr 27 11:47:47.389523 Resolve network interfaces

Sessions                 1 (1 PPPoE / 0 IPoE)
  Established           0 [                                                            ]
  Outstanding          0 [                                                            ]
  Terminated           0 [                                                            ]
  Setup Time           0 ms
  Setup Rate            0.00 CPS (MIN: 0.00 AVG: 0.00 MAX: 0.00)
  Flapped                 0

Traffic Flows Verified
  Session            0/2 [                                                            ]

Network Interface ( vlan90-if:90 )
  TX Packets                                  13 |      1 PPS          0 Kbps
  RX Packets                                   0 |      0 PPS          0 Kbps
  TX Session Packets IPv4            0 |      0 PPS
  RX Session Packets IPv4            0 |      0 PPS          0 Loss
  TX Session Packets IPv6            0 |      0 PPS
  RX Session Packets IPv6            0 |      0 PPS          0 Loss
  TX Session Packets IPv6PD       0 |      0 PPS
  RX Session Packets IPv6PD       0 |      0 PPS          0 Loss
  TX Multicast Packets                  0 |      0 PPS

Access Interface ( vlan100-if )
  TX Packets                                   0 |      0 PPS          0 Kbps
  RX Packets                                   0 |      0 PPS          0 Kbps
  TX Session Packets IPv4            0 |      0 PPS
  RX Session Packets IPv4            0 |      0 PPS          0 Loss 0 Wrong Session
  TX Session Packets IPv6            0 |      0 PPS
  RX Session Packets IPv6            0 |      0 PPS          0 Loss 0 Wrong Session
  TX Session Packets IPv6PD       0 |      0 PPS
  RX Session Packets IPv6PD   
  
  
  
  Can you help with network interface configuration. 
  -->The ip and gateway can be set randomly of our choice?  
  -->Is gateway a physical system? where to get Mac of gateway if I keep getting arp request for it.

**Comment by GIC-de** [comment, marked as answer]
The topology should be like:

BNG Blaster Access Interface === vBNG === BNG Blaster Network Interface

The network interface should be connected to an IP interface of your virtual BNG. The gateway IP should be the IP of the virtual BNG. If both interfaces are there, the Blaster is able to generate traffic between access session interface and the IP addresses of the network interface.

**Reply by vbhurangi** [reply]
Thankyou, I will try it out

**Reply by vbhurangi** [reply]
Hi Christian,
Still I was not able to get traffic through.

I have connected network interface directly to vBNG interface(ge-0/0/2). Assigned ip and vlan on logical interface ge-0/0/2.0 .

Config for network interface  on bngblaster:
address- IP address of logical interface ge-0/0/2.0 
gateway- IP address of vBNG.
vlan- vlan assigned to ge-0/0/2.0 

address and gateway are kept in same domain.

The traffic does not seem to reach the vBNG ge-0/0/2 or ge-0/0/2.0 interface .

Can you please suggest where I can correct or check my config?
